BUILD.gn

```diff
@@ -41,6 +41,18 @@ if (chip_with_lwip) {
   import("//build_overrides/lwip.gni")
 }
 
+config("pigweed_wno_errors_config") {
+  cflags = [
+    "-Wno-error=undef",
```

Comment:
I side with @cecille. Undef error was specifically added recently to the whole repo by @bzbarsky-apple.
Those error no-error flags apply to the matter test suite and unit test.

In what location do you get the errors? Either try to fix the error in the code or at least try to apply the cflag to a narrower set of source/third_party if you can't fix them directly.

Reply:
@bzbarsky-apple  I think that you are right. This configuration is used as `public_deps` in pigweed modules and it will be shared with direct dependencies [gn public_deps](https://gn.googlesource.com/gn/+/main/docs/reference.md#var_public_deps). 
I've checked it for the prepared test in #29682. Unfortunately, during the compilation of `TestAccessControl.cpp`, these flags are also there.

Reply:
Yeah, this is the big problem with pigweed: right now we have to disable lots of warnings in any build that uses pigweed, because it's full of warnings and a bunch of those warnings are in headers.  :(

Reply:
Ah, the headers. Does this turn off warnings across the board for all code if it's compiled with pigweed? Is there maybe a way to reduce the impact zone of this?

@andy31415 - do you know if the pigweed folks know about this? The real solution is to get them to fix this.